tests/src/FunctionalJavascript/CollaboraModalFormatterTest.php

```diff
@@ -0,0 +1,315 @@
+   *   Expected dialog width in px, or NULL for auto-sized width.
+   *
+   * @return \Behat\Mink\Element\NodeElement
+   *   The iframe element.
+   */
+  protected function assertModalWithIframe(MediaInterface $media, int|null $expected_dialog_width = NULL): NodeElement {
+    $assert_session = $this->assertSession();
+    $iframe = $assert_session->waitForElementVisible('css', '.ui-dialog.cool-modal-preview > .ui-dialog-titlebar + .ui-dialog-content > iframe.cool-iframe');
+    $this->assertSame('/cool/view/' . $media->id(), $iframe->getAttribute('src'));
+
+    [
+      $viewport_width,
+      $viewport_height,
+      $dialog_rect,
+      $titlebar_rect,
+      $content_rect,
+      $iframe_rect,
+    ] = $this->getSession()->evaluateScript(<<<JS
+[
+  window.innerWidth,
+  window.innerHeight,
+  document.querySelector('.ui-dialog.cool-modal-preview').getBoundingClientRect(),
+  document.querySelector('.ui-dialog-titlebar').getBoundingClientRect(),
+  document.querySelector('.ui-dialog-content').getBoundingClientRect(),
+  document.querySelector('iframe.cool-iframe').getBoundingClientRect(),
+]
+JS);
+    $this->assertNoJsConsoleLogs();
+
+    if ($expected_dialog_width !== NULL) {
+      // The dialog has the expected width.
+      // Allow for padding and 'box-sizing: content-box'.
+      $this->assertNumberInRange($expected_dialog_width, 12, $dialog_rect['width']);
+    }
+    else {
+      // The dialog width is relative to the viewport.
+      // Allow for padding and 'box-sizing: content-box'.
+      $this->assertNumberInRange($viewport_width * .92 - 1, 12, $dialog_rect['width']);
+    }
+    // The dialog height is a percentage of the viewport height.
+    // Allow for padding and 'box-sizing: content-box'.
+    $this->assertLessThanOrEqual($viewport_height - 20, $dialog_rect['height']);
+    $this->assertGreaterThanOrEqual($viewport_height * .8, $dialog_rect['height']);
+
+    // The dialog is centered in the viewport.
+    $this->assertEqualsWithDelta($dialog_rect['left'], $viewport_width - $dialog_rect['right'], 2);
+    $this->assertEqualsWithDelta($dialog_rect['top'], $viewport_height - $dialog_rect['bottom'], 2);
+
+    // Assert horizontal positioning of child elements.
+    $this->assertNumberInRange($dialog_rect['left'], 6, $content_rect['left']);
+    $this->assertNumberInRange($dialog_rect['right'], -6, $content_rect['right']);
+    $this->assertSame($content_rect['left'], $titlebar_rect['left']);
+    $this->assertSame($content_rect['right'], $titlebar_rect['right']);
+
+    // Assert vertical positioning of child elements.
+    $this->assertNumberInRange($dialog_rect['top'], 6, $titlebar_rect['top']);
+    $this->assertNumberInRange($titlebar_rect['bottom'], 4, $content_rect['top']);
+    $this->assertNumberInRange($dialog_rect['bottom'], -6, $content_rect['bottom']);
+
+    // The iframe has the same dimensions and position as its parent.
+    $this->assertSame($content_rect, $iframe_rect);
```

Comment:
As I commented in the previous PR, it's not the job of this type of test to verify the rendering of the modal with these precise assertions. Let's delete all of this.

Reply:
We do not currently have different type of test easily available.
I don't want to introduce a different test framework or tool.

The above assertions will fail if we remove or damage the CSS, which is exactly why I added them.
If somebody thinks they have a great idea to refactor the CSS or replace it with a different mechanism, they need to attempt to make these tests pass.

They are precise, but they are still permissive towards differences from the theme.

Reply:
Btw in core I find three places with getBoundingClientRect used in php tests:
- CKEditor5ToolbarTest
- LayoutBuilderDisableInteractionsTest
- JSWebAssert

My approach would be:
- Do we want to know if we have a regression in the element positioning, e.g. after a theming refactor?
- Is there a better way to achieve this, without introducing a new test framework?